src/pages/transaction/index.tsx

```diff
@@ -74,11 +75,19 @@ export const TransactionPage: React.FC = () => {
     return 'Sign transaction';
   }, [txType]);
 
+  useEffect(() => {
+    window.addEventListener('beforeunload', async function () {
+      if (!isSubmitting) {
+        await doSubmitOrCancelPendingTransaction('cancel');
+      }
+    });
+  }, [isSubmitting, doSubmitOrCancelPendingTransaction]);
+
```

Comment:
You should return the removeEventListener, too, eg:
```suggestion
const handleUnmount = async () => {
  if (!isSubmitting) {
    await doSubmitOrCancelPendingTransaction('cancel');
  }
}
  useEffect(() => {
    window.addEventListener('beforeunload', handleUnmount);
    return () => window.removeEventListener('beforeunload', handleUnmount);
  }, [isSubmitting, doSubmitOrCancelPendingTransaction]);

```

Reply:
O yeah, good catch, thanks!